source/vibe/aws/s3.d

```diff
@@ -154,7 +154,7 @@ class S3 : RESTClient
     this(string bucket, string region, AWSCredentialSource credsSource, ClientConfiguration config = ClientConfiguration())
     {
         this.bucket = bucket;
-        super(bucket ~ ".s3-" ~ region ~ ".amazonaws.com", region, "s3", credsSource, config);
+        super(bucket ~ (region ? ".s3-" ~ region ~ ".amazonaws.com" : ".s3.amazonaws.com"), region, "s3", credsSource, config);
```

Comment:
we should at least check for non empty string in the constructors pre-condition and remove non-region check.
see also https://github.com/tamediadigital/vibe-s3/issues/5